extensions/amp-bind/0.1/bind-expr-impl.jison

```
@@ -0,0 +1,288 @@
+    '!' expr
+      {$$ = !$2;}
+  | '-' expr %prec UMINUS
+      {$$ = -$2;}
+  | '+' expr %prec UPLUS
+      {$$ = +$2;}
+  |  expr '+' expr
+      {$$ = $1 + $3;}
+  | expr '-' expr
+      {$$ = $1 - $3;}
+  | expr '*' expr
+      {$$ = $1 * $3;}
+  | expr '/' expr
+      {$$ = $1 / $3;}
+  | expr '%' expr
+      {$$ = $1 % $3;}
+  | expr '&&' expr
+      {$$ = $1 && $3;}
+  | expr '||' expr
+      {$$ = $1 || $3;}
+  | expr '<=' expr
+      {$$ = $1 <= $3;}
+  | expr '<' expr
+      {$$ = $1 < $3;}
+  | expr '>=' expr
+      {$$ = $1 >= $3;}
+  | expr '>' expr
+      {$$ = $1 > $3;}
+  | expr '!=' expr
+      {$$ = $1 != $3;}
+  | expr '==' expr
+      {$$ = $1 == $3;}
+  | expr '?' expr ':' expr
+      {$$ = $1 ? $3 : $5;}
+  ;
+
+invocation:
+    expr '.' NAME args
+      %{
+        $$ = null;
+
+        var obj = Object.prototype.toString.call($1);
+
+        var whitelist = functionWhitelist[obj];
+        if (whitelist) {
+          var fn = $1[$3];
+          if (whitelist.indexOf(fn) >= 0) {
+            if (typeCheckArgs($4)) {
+              $$ = fn.apply($1, $4);
+            } else {
+              throw new Error(`Unexpected argument type in {$3}()`);
+            }
+            return;
+          }
+        }
+
+        throw new Error(`{$3}() is not a supported function.`);
+      %}
+  ;
+
+args:
+    '(' ')'
+      {$$ = [];}
+  | '(' array ')'
+      {$$ = $2;}
+  ;
+
+member_access:
+    expr member
+      %{
+        $$ = null;
+
+        if ($1 === null || $2 === null) {
+          return;
+        }
+
+        var prop = Object.prototype.toString.call($2);
+        if (prop === '[object String]' || prop === '[object Number]') {
+          if (Object.prototype.hasOwnProperty.call($1, $2)) {
+            $$ = $1[$2];
+          }
+        }
+      %}
+  ;
+
+member:
+    '.' NAME
+      {$$ = $2;}
+  | '[' expr ']'
+      {$$ = $2;}
+  ;
+
+variable:
+    NAME
+      {$$ = Object.prototype.hasOwnProperty.call(yy, $1) ? yy[$1] : null;}
```

It should throw an error.